### PR TITLE
Several TSLint fixes

### PR DESCRIPTION
**Pull request by mgolawski**
Proposition of changing several TSLint rules, for better code maintainability:

`jsx-no-lambda` - I will leave current setup with *TODO* mark, for fixes in next pull request. Allowing such code structure in components renders may lead to memory leaks and performance hiccups, and there is no gain from that.
`no-trailing-whitespace` - this should be restricted
 `object-literal-sort-keys` - i propose to set up this rule as *warning* and set as `match-declaration-order`, `shorthand-first`. Firstly it helps with merge conflicts, and code maintainability, and enforces contributors to type their code.
`no-submodule-imports` - there is no point to allow that. I fix the invalid imports already.
`no-implicit-dependencies` - for now i would disable it, but in the future it may cause some leftovers in the code. Right now i can't fix because of submodule architecture.
`no-object-literal-type-assertion` - fixed invalid code, and get rid of that.
`no-unused-expression` - fixed invalid code, and get rid of that.

In the next few PR's i propose to add:
- Husky for TSLing and Prettier hooks, to avoid commiting invalid code to master.
- Adds CI for PR status checks for code coverage, unit tests, and linters.
- Rename `tsx` to `ts` files for non-react code.

Let me know what do you think about that.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,33 +24,35 @@
     "react-native-image-picker": "^0.28.1",
     "react-native-splash-screen": "^3.2.0",
     "react-native-vector-icons": "^6.5.0",
+    "react-native": "0.59.9",
     "react-navigation": "^3.11.0",
-    "yup": "^0.27.0"
+    "react": "16.8.6",
+    "yup": "^0.27.0",
+    "enzyme": "^3.10.0"
```
Comment (bklukaczewski) [inline package.json]: Isn't enzyme dev only dependency?

File: package.json [hunk 2]
```diff
@@ -26,33 +24,35 @@
     "react-native-image-picker": "^0.28.1",
     "react-native-splash-screen": "^3.2.0",
     "react-native-vector-icons": "^6.5.0",
+    "react-native": "0.59.9",
```
Comment (bklukaczewski) [inline package.json]: Can we stick to alphabetical order for dependencies?

File: src/screens/DialogScreen.tsx [hunk 3]
```diff
@@ -32,7 +31,7 @@ export class DialogScreen extends React.PureComponent<
   state = {
     title: '',
     description: '',
-    onPress: emptyFunction,
+    onPress: () => null,
```
Comment (bklukaczewski) [inline src/screens/DialogScreen.tsx]: As far I remember it was done that way so there were no issues with code coverage reports

File: src/styles/typography.tsx [hunk 4]
```diff
@@ -36,7 +58,6 @@ export const typography = StyleSheet.create({
   // used for header text, some bigger inputs
   headline6: {
     fontFamily: fonts.sansSerif.medium,
-    fontWeight: undefined,
```
Comment (bklukaczewski) [inline src/styles/typography.tsx]: It was needed as default React Navigation header `fontWeight` was `500` (or iOS or Android, can't remember), but it didn't played well with custom fonts.

**Comment by mgolawski** [comment]
Hey. I added the code review fixes, and did a merge with @mikolevy changes. Can you check it out again? Thanks!

File: package.json (hunk 2 above)
Comment (mgolawski) [reply package.json]: Well, it is how sorting extension from VSCode did this.

File: src/styles/typography.tsx (hunk 4 above)
Comment (mgolawski) [reply src/styles/typography.tsx]: I set it to the `400` then, instead of undefined.

File: package.json (hunk 1 above)
Comment (mgolawski) [reply package.json]: Right, changed.

File: src/screens/DialogScreen.tsx (hunk 3 above)
Comment (mgolawski) [reply src/screens/DialogScreen.tsx]: I will check it out later. Maybe we should use `lodash.noop` then.

File: package.json (hunk 2 above)
Comment (bklukaczewski) [reply package.json]: They use different alphabet then ;) If you do `yarn add dependency` it follows the rule, that "react-native" should be before any "react-native-X"

**Pull request merged**